google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -187,9 +186,9 @@ public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
     // Handle schema updates in a Thread-safe way by locking down the operation
     synchronized (this) {
       // Update schema only work when connection pool is not enabled.
-      if (this.streamWriter.getConnectionOperationType() == Kind.CONNECTION_WORKER
-          && this.streamWriter.getUpdatedSchema() != null) {
-        refreshWriter(this.streamWriter.getUpdatedSchema());
+      TableSchema updatedSchema = this.streamWriter.getUpdatedSchema();
+      if (updatedSchema != null && !this.tableSchema.equals(updatedSchema)) {
+        refreshWriter(updatedSchema);
```

Comment:
Per discussion, we only need to update the tableSchema and be sure to send in the new schema, without needing to reconnect?

Reply:
nvm.